tests/ui/did_you_mean/casting-fn-item-to-fn-pointer.rs

```diff
@@ -0,0 +1,8 @@
+//@ edition: 2021
```

Comment:
why does this test rely on an edition?

Reply:
for the original case in the issue, we cannot get the same TypeError in edition 2018, but I found the two cases in the test can get same TypeError in different editions.

I have updated the test with the original case, which cannot pass in edition 2018